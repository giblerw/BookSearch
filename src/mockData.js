export const emptySet = {
    currentQuery: 'default',
    listData: {},
    totalItems: 0,
    activeSort: 'relevance',
}

export const set1 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: ['Vera Brittain'],
            description: `This classic memoir of the First World War is now a major motion picture starring Alicia Vikander and Kit Harington. Includes an afterword by Kate Mosse OBE. In 1914 Vera Brittain was 20, and as war was declared she was preparing to study at Oxford. Four years later her life - and the life of her whole generation - had changed in a way that would have been unimaginable in the tranquil pre-war era. TESTAMENT OF YOUTH, one of the most famous autobiographies of the First World War, is Brittain's account of how she survived those agonising years; how she lost the man she loved; how she nursed the wounded and how she emerged into an altered world. A passionate record of a lost generation, it made Vera Brittain one of the best-loved writers of her time, and has lost none of its power to shock, move and enthral readers since its first publication in 1933.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const set2 = {
    items: [{
        volumeInfo: {
            title: 'Hoot',
            authors: ['Carl Hiaasen'],
            description: `Roy, who is new to his small Florida community, befriends a mysterious boy who introduces him to potty-trained alligators, burrowing owls, and sparkly-tailed poisonous snakes. A Newbery Honor Book, ALA Notable Book, & ALA Best Book for Young Adults. Reissue.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=BTDSfx5bunwC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
};

export const set3 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: null,
            description: '',
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const beforeSearch = {
    query: '',
    searched: false
}

export const afterSearch = {
    query: 'harry',
    searched: true
}