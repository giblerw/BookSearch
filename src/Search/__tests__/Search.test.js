import { mount } from '@vue/test-utils';
import SearchComponent from '../index.vue';

import { beforeSearch, afterSearch } from '../../mockData';

describe('Search.vue', () => {

    it('renders the expected HTML structure, before search', () => {
        const wrapper = mount(SearchComponent, {
            data: () => beforeSearch 
        });
        expect(wrapper.element).toMatchSnapshot();
    })

    it('renders the expected HTML structure, after search', () => {
        const wrapper = mount(SearchComponent, {
            data: () => afterSearch 
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});