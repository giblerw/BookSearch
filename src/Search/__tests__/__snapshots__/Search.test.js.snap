// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue renders the expected HTML structure, after search 1`] = `
<div
  class="search-bar-container"
>
  <form>
    <input
      placeholder="Search here"
      type="search"
    />
     
    <i
      class="fa fa-search"
    >
      <button
        id="search-submit"
        type="submit"
      />
    </i>
  </form>
   
  <span
    id="sort-button-group"
  >
    <span>
      Sort By: 
    </span>
     
    <button
      class="relevant-sort-btn active"
      value="relevance"
    >
      
        Relevance
    
    </button>
     
    <button
      class="newest-sort-btn"
      disabled="disabled"
      value="newest"
    >
      
        Newest
    
    </button>
  </span>
</div>
`;

exports[`Search.vue renders the expected HTML structure, before search 1`] = `
<div
  class="search-bar-container"
>
  <form>
    <input
      placeholder="Search here"
      type="search"
    />
     
    <i
      class="fa fa-search"
    >
      <button
        id="search-submit"
        type="submit"
      />
    </i>
  </form>
   
  <!---->
</div>
`;
